# Assignment: Explore Loops in Python

**Hey there, Python enthusiast!**  
Get ready to dive deep into loops and unleash their full power! In this assignment, you’ll practice using both `for` and `while` loops to solve practical problems. Let’s get started!

**Task 1 - Counting Down with Loops**  
Write a Python program to create a countdown timer.

1.  Ask the user for a starting number.
2.  Use a `while` loop to print numbers from that number down to 1.
3.  When the countdown ends, print a celebratory message like "Blast off!"

For example:

`Enter the starting number: 5`

`5 4 3 2 1 Blast off! 🚀`

In [5]:
num_count = int(input("Input a starting number please!"))

while num_count >= 1:
    print(str(num_count) + " ", end = "")

    num_count -= 1

print("Blastoff! 🚀")

5 4 3 2 1 Blastoff! 🚀


**Task 2 - Multiplication Table with `for` Loops**  
Write a program that generates the multiplication table for any number provided by the user.

1.  Ask the user to input a number.
2.  Use a `for` loop to print the multiplication table for that number (from 1 to 10).

Example Output:

`Enter a number: 4`

`4 x 1 = 4 4 x 2 = 8 ... 4 x 10 = 40`



In [10]:
num_count2 = int(input("Please input a number please"))

for i in range(1, 10 + 1):
    print(str(num_count2), "x", str(i), "=", str(num_count2 * i), end = " ")

20 x 1 = 20 20 x 2 = 40 20 x 3 = 60 20 x 4 = 80 20 x 5 = 100 20 x 6 = 120 20 x 7 = 140 20 x 8 = 160 20 x 9 = 180 20 x 10 = 200 

**Task 3 - Find the Factorial**  
Write a Python program to calculate the factorial of a number entered by the user.

1.  Ask the user for a number.
2.  Use a `for` loop to calculate the factorial.
3.  Print the result in a friendly format.

For example:

`Enter a number: 5`

`The factorial of 5 is 120.`

In [14]:
num_count3 = int(input("Please input a number please"))

# Calculating factorial with user input
for i in range(1, num_count3):
    num_count3 *= i # num_count3 already the highest number, 
                    # so we start from 1 and multiply it by the next number until we reach the number before the user input

print(num_count3)

120
